numpy/f2py/backends/meson_backend.py

```diff
@@ -0,0 +1,183 @@
+from __future__ import annotations
+
+import os
+import errno
+import shutil
+import subprocess
+from pathlib import Path
+
+from .backend import Backend
+from string import Template
+
+class MesonTemplate:
+	"""Template meson build file generation class."""
+	def __init__(self, module_name: str, numpy_install_path, numpy_get_include: Path, f2py_get_include: Path, wrappers: list[Path], fortran_sources: list[Path], dependencies: list[str], include_path: list[Path], optimization_flags: list[str], architecture_flags: list[str], f77_flags: list[str], f90_flags: list[str], linker_libpath: list[Path], linker_libname: list[str], define_macros: list[tuple[str, str]], undef_macros: list[str]):
+		self.module_name = module_name
+		self.numpy_install_path = numpy_install_path
+		self.build_template_path = numpy_install_path / "f2py" / "backends" / "src" / "meson.build.src"
+		self.sources = fortran_sources
+		self.numpy_get_include = numpy_get_include
+		self.f2py_get_include = f2py_get_include
+		self.wrappers = wrappers
+		self.dependencies = dependencies
+		self.include_directories = include_path
+		self.substitutions = {}
+		self.optimization_flags = optimization_flags
+		self.architecture_flags = architecture_flags
+		self.fortran_flags = f77_flags + f90_flags
+		self.linker_libpath = linker_libpath
+		self.linker_libname = linker_libname
+		self.define_macros = define_macros
+		self.undef_macros = undef_macros
+		self.pipeline = [self.initialize_template,
+						self.global_flags_substitution,
+                   		self.sources_substitution,
+                      	self.dependencies_substitution,
+                       	self.include_directories_subtitution,
+                        self.linker_substitution,
+                        self.macros_substitution]
+
+	@property
+	def meson_build_template(self) -> str:
+		if(not self.build_template_path.is_file()):
+			raise FileNotFoundError(errno.ENOENT, f"Meson build template {self.build_template_path.absolute()} does not exist.")
+		return self.build_template_path.read_text()
+
+	def initialize_template(self) -> None:
+		"""Initialize with module name and external NumPy and F2PY C libraries."""
+		self.substitutions['modulename'] = self.module_name
+		self.substitutions['numpy_get_include'] = self.numpy_get_include.absolute()
+		self.substitutions['f2py_get_include'] = self.f2py_get_include.absolute()
```

Comment:
Are these necessary? The [documentation includes](https://numpy.org/doc/stable/f2py/buildtools/meson.html) the option to get these dynamically from `meson`...